src/System.Private.CoreLib/shared/System/Reflection/TypeInfo.cs

```diff
@@ -15,11 +16,11 @@ public abstract partial class TypeInfo : Type, IReflectableType
 
         public virtual Type[] GenericTypeParameters => IsGenericTypeDefinition ? GetGenericArguments() : Type.EmptyTypes;
 
-        public virtual EventInfo GetDeclaredEvent(string name) => GetEvent(name, TypeInfo.DeclaredOnlyLookup);
-        public virtual FieldInfo GetDeclaredField(string name) => GetField(name, TypeInfo.DeclaredOnlyLookup);
+        public virtual EventInfo? GetDeclaredEvent(string name) => GetEvent(name, TypeInfo.DeclaredOnlyLookup);
+        public virtual FieldInfo? GetDeclaredField(string name) => GetField(name, TypeInfo.DeclaredOnlyLookup);
         public virtual MethodInfo GetDeclaredMethod(string name) => GetMethod(name, TypeInfo.DeclaredOnlyLookup);
```

Comment:
GetDeclaredMethod can also return null...  https://source.dot.net/#System.Private.CoreLib/src/System/RtType.cs,2731

Reply:
RuntimeType and TypeInfo are overriding that method from MethodInfo class, which is not included in this PR and am working on it right now, so for now these methods nullable annotations not affecting each other

Reply:
with that being said i mean its coming with anotations MethodInfo and its descendants